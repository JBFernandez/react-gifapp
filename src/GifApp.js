import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifApp = () => {

   
    const [categories, setCategories] = useState(["Dragon Ball" ]);

    /* const handleAdd = () => {
        setCategories( [...categories, "MonsterHunter"] );        
        setCategories( cats => [...cats, "MonsterHunter"] );        
         } */    

    return (
        <>
            <h2> GifApp </h2>
            < AddCategory setCategories={ setCategories }/> {/* anything we send here is received as a props */}
            <hr/> {/*  REACT tag to place a black line */}        
            <ol>
                {
                    categories.map( category =>  (
                        <GifGrid 
                            key={ category }
                            category={ category }  /* the propp category is the category iterated category of categories */
                        /> 
                        ))
                }
            </ol>
        </>
    );
}

export default GifApp;