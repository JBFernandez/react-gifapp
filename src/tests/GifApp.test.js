import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifApp from '../GifApp';


describe('Tests on <GifApp/>', () => { 


    test('should display the component and match with the snapshot ', () => { 

        const wrapper = shallow( <GifApp/> );
        expect( wrapper ).toMatchSnapshot();
     });

     test('should diplay a list of categories', () => { 
         
        const categories = ['LOTR', 'Dragon Ball'];
        const wrapper = shallow( <GifApp defaultCategories={categories} /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

      })

});