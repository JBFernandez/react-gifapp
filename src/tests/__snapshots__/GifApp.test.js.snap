// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifApp/> should diplay a list of categories 1`] = `
<Fragment>
  <h2>
     GifApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
   
  <hr />
   
  <ol>
    <GifGrid
      category="LOTR"
      key="LOTR"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Tests on <GifApp/> should display the component and match with the snapshot  1`] = `
<Fragment>
  <h2>
     GifApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
   
  <hr />
   
  <ol />
</Fragment>
`;
